packages/runtime-core/src/apiWatch.ts

```diff
@@ -179,11 +179,11 @@ function doWatch(
       // immediately watch or watchEffect
       baseWatchOptions.once = true
     } else {
-      return {
-        stop: NOOP,
-        resume: NOOP,
-        pause: NOOP,
-      } as WatchHandle
+      const watchStopHandle = () => {}
+      watchStopHandle.stop = NOOP
+      watchStopHandle.resume = NOOP
+      watchStopHandle.pause = NOOP
+      return watchStopHandle
```

Comment:
I considered doing this:

```ts
return Object.assign(() => {}, {
  stop: NOOP,
  resume: NOOP,
  pause: NOOP,
})
```

But decided against it to avoid the temporary allocation of the object literal.